libflux/src/core/ast/mod.rs

```diff
@@ -492,8 +492,131 @@ pub struct BuiltinStmt {
     #[serde(flatten)]
     pub base: BaseNode,
     pub id: Identifier,
+    pub _type: TypeExpression,
 }
 
+#[derive(Debug, PartialEq, Clone, Serialize, Deserialize)]
+pub enum MonoType {
+    Tvar(TvarType),
+    Int(Int),
+    Uint(Uint),
+    Float(Float),
+    String(StringType),
+    Bool(Bool),
+    Time(Time),
+    Duration(DurationType),
+    Bytes(Bytes),
+    Regexp(Regexp),
+    Invalid,
```

Comment:
I'd like to see these names use a consistent pattern. Should we use the `Type` suffix for all of the types?

Meaning like this:

```
pub enum MonoType {
    Tvar(TvarType),
    Int(IntType),
    Uint(UintType),
    Float(FloatType),
    String(StringType),
    Bool(BoolType),
    Time(TimeType),
    Duration(DurationType),
    Bytes(BytesType),
    Regexp(RegexpType),
    Invalid,
}
```